src-js/optimizelyClient.js

```diff
@@ -7,6 +7,17 @@ class OptimizelyClient {
       datafile: window.optimizelyDatafile,
     });
   }
+  getUserId() {
+    return new Promise((resolve) => {
+      if (window.userData) {
+        resolve((window.userData.analytics_id) ? window.userData.analytics_id : null);
+      } else {
+        window.addEventListener("userDataReady", () => {
+          resolve((window.userData.analytics_id) ? window.userData.analytics_id : null);
```

Comment:
could we include a quick comment for why we are making this check? I think what you listed in **Reason** would be perfect!

Reply:
Good catch! Done ✅